File: python/examples/streaming/basic_call.py
from mirascope import llm


@llm.call("openai/gpt-5-mini")
def recommend_book(genre: str):
    return f"Recommend a {genre} book."


response: llm.StreamResponse = recommend_book.stream("fantasy")
for chunk in response.text_stream():
    print(chunk, end="", flush=True)

File: typescript/examples/streaming/basic-call.ts
import { llm } from "mirascope";

const recommendBook = llm.defineCall<{ genre: string }>({
  model: "openai/gpt-5-mini",
  template: ({ genre }) => `Recommend a ${genre} book`,
});

const response = await recommendBook.stream({ genre: "fantasy " });

process.stdout.write("Streaming: ");
for await (const text of response.textStream()) {
  process.stdout.write(text);
}
console.log("\n");
